kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_with_constraints.h

```diff
@@ -0,0 +1,1288 @@
+        const auto constraints_begin = rModelPart.MasterSlaveConstraintsBegin();
+        #pragma omp parallel for schedule(guided, 512) firstprivate(n_constraints, constraints_begin)
+        for (int k = 0; k < n_constraints; ++k) {
+            auto it = constraints_begin + k;
+            it->FinalizeSolutionStep(r_process_info);
+        }
+        KRATOS_CATCH("ResidualBasedEliminationBuilderAndSolverWithConstraints failed to finalize solution step.")
+    }
+
+    ///@}
+    ///@name Access
+    ///@{
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const override
+    {
+        return "ResidualBasedEliminationBuilderAndSolverWithConstraints";
+    }
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const override
+    {
+        rOStream << Info();
+    }
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const override
+    {
+        rOStream << Info();
+    }
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+    ///@}
+protected:
+    ///@name Protected static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+    /**
+    * @brief This method assembles the global relation matrix
+    * @param rT The global relation matrix
+    * @param rTransformationMatrix The local transformation contribution
+    * @param rSlaveEquationId The equation id of the slave dofs
+    * @param rMasterEquationId The equation id of the master dofs
+    * @param rLockArray The lock of the dof
+    */
+    void AssembleRelationMatrix(
+        TSystemMatrixType& rT,
+        const LocalSystemMatrixType& rTransformationMatrix,
+        const EquationIdVectorType& rSlaveEquationId,
+        const EquationIdVectorType& rMasterEquationId
+    #ifdef _OPENMP
+        ,std::vector< omp_lock_t >& rLockArray
+    #endif
+        )
+    {
+        SizeType local_size_1 = rTransformationMatrix.size1();
+        SizeType local_size_2 = rTransformationMatrix.size2();
+
+        for (IndexType i_local = 0; i_local < local_size_1; ++i_local) {
+            IndexType i_global = rSlaveEquationId[i_local];
+
+            if (i_global < BaseType::mEquationSystemSize) {
+            #ifdef _OPENMP
+                omp_set_lock(&rLockArray[i_global]);
+            #endif
+                for (IndexType j_local = 0; j_local < local_size_2; ++j_local) {
+                    IndexType j_global = mSolvableDoFReorder[rMasterEquationId[j_local]];
+                    if (j_global < BaseType::mEquationSystemSize) {
+                        rT(i_global, j_global) += rTransformationMatrix(i_local, j_local);
+                    }
+                }
+            #ifdef _OPENMP
```

Comment:
get rid

Reply:
OK

Reply:
I cannot, problems with acceding the sparse_matrix components